src/h2olog/h2olog.cc

```diff
@@ -251,6 +247,9 @@ int main(int argc, char **argv)
     pid_t h2o_pid = -1;
     while ((c = getopt(argc, argv, "hdrlp:t:s:w:")) != -1) {
         switch (c) {
+        case 'H':
+            tracer.reset(create_http_tracer());
```

Comment:
I'm not sure if this is going to work, as `tracer` is used before reaching this line, when `tracer->udst_probes()` is being called (see above).

I tend to assume that `-t` cannot be used in conjunction with `-H`, and that use of `-t` should be forbidden when `-H` is being set. WDYT?

Reply:
Right. It leads to behaviors that will surprise users. 

Because each tracer knows what kind of customizations are available, so I'll move that logic to tracers: 3401254

It also hides the implementation details of "available USDTs" and "selected USDTs", making the semantics of `-l` to be changed from "list of USDTs to be available" to "list of USDTs to be used". I'll move the logic of `-l` out of switch blocks for `-l` not to depends on the order of options.